zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala

```diff
@@ -240,6 +240,22 @@ class IncrementalCompilerImpl extends IncrementalCompiler {
         case Some(previous) => previous
         case None           => Analysis.empty
       }
+
+      val compileStraightToJar = STJ.isEnabled(output)
```

Comment:
Are there any unit tests for straight to jar behaviors?

Reply:
There are no unit tests, I was relying on scripted tests. I will take a look on what should potentially be tested.

Reply:
scripted tests are fine too.